### PR TITLE
Properly remove the service container during kube down

### PR DESCRIPTION
**Pull request by umohnani8**
Ensure that we are actually looking up the service container ID and actually removing it during kube teardown for the --wait use case. This ensures that we don't have a service container waiting around in removing state before we return from kube play in the remote case.

Should fix https://github.com/containers/podman/issues/17803

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by umohnani8** [comment]
@vrothberg PTAL

**Review by vrothberg — APPROVED** [review]
LGTM

**Comment by rhatdan** [comment]
LGTM
Other then the tests failing.

File: pkg/domain/infra/abi/play.go [hunk 1]
```diff
@@ -1343,6 +1343,20 @@ func (ic *ContainerEngine) PlayKubeDown(ctx context.Context, body io.Reader, opt
...
 
+	// Get the service containers associated with the pods if any
+	serviceCtrIDs := []string{}
+	for _, name := range podNames {
+		pod, err := ic.Libpod.LookupPod(name)
+		if err != nil {
+			return nil, err
+		}
+		ctr, err := pod.ServiceContainer()
+		if err != nil {
```
Comment (vrothberg) [inline pkg/domain/infra/abi/play.go:1354]: Needs a
```golang
if errors.Is(err, define.ErrNoSuchCtr) {
    continue
}
```

Comment (umohnani8) [reply pkg/domain/infra/abi/play.go:1357]: fixed

**Comment by TomSweeneyRedHat** [comment]
LGTM and happy green test buttons.
Kicking it down the road.

**Pull request merged**